flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/ZooKeeperCheckpointIDCounter.java

```diff
@@ -132,6 +132,17 @@ public long getAndIncrement() throws Exception {
 		}
 	}
 
+	@Override
+	public long get() {
+		ConnectionState connState = connStateListener.getLastState();
+
+		if (connState != null) {
```

Comment:
Whenever you feel tempted to copy code from a method, maybe think about extracting it into a common helper. This pattern get & check is just repeated from `getAndIncrement`